src/pybind/mgr/prometheus/module.py

```diff
@@ -1033,6 +1033,8 @@ def get_pg_status(self) -> None:
 
         for pool in pg_summary['by_pool']:
             num_by_state: DefaultDict[str, int] = defaultdict(int)
+            for state in PG_STATES:
+                num_by_state[state] = 0
```

Comment:
This adds [33](https://github.com/rhcs-dashboard/ceph/blob/wip-58502-pacific/src/pybind/mgr/mgr_module.py#L46-L79) x `len(pools)` data samples per scrape. Nothing dramatic, but it grows with the number of pools. Just to keep it in mind.

Reply:
Yeah , I was considering it, but given there's the requirement for these metrics even with zero values I added initialization again.